test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ValidateTests.cs

```diff
@@ -0,0 +1,170 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Xunit;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest
+{
+    [Trait(Traits.Category, Categories.Validate)]
+    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.Json)]
+    public class ValidateTests : IClassFixture<HttpIntegrationTestFixture>
+    {
+        private readonly Common.FhirClient _client;
+
+        public ValidateTests(HttpIntegrationTestFixture fixture)
+        {
+            _client = fixture.FhirClient;
+        }
+
+        [Theory]
+        [InlineData("Patient/$validate", "{\"resourceType\":\"Patient\",\"name\":{\"family\":\"test\",\"given\":\"one\"}}")]
+        [InlineData("Observation/$validate", "{\"resourceType\":\"Observation\",\"status\":\"registered\",\"code\":{\"coding\":[{\"system\":\"system\",\"code\":\"code\"}]}}")]
+        public async void GivenAValidateRequest_WhenTheResourceIsValid_ThenAnOkMessageIsReturned(string path, string payload)
+        {
+            OperationOutcome outcome = await _client.ValidateAsync(path, payload);
+
+            Assert.Single(outcome.Issue);
+            CheckOperationOutcomeIssue(
+                outcome.Issue[0],
+                OperationOutcome.IssueSeverity.Information,
+                OperationOutcome.IssueType.Informational,
+                "All OK");
```

Comment:
Maybe a private string const since it's used in multiple places in this file?

Reply:
Changed